src/python/pants/backend/python/rules/targets.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
We could use the V1 style of having a dedicated file for each target type. But, Benjy and I weren't in love with that pattern because it results in so many files and discourages reuse of fields.

A key difference from V1 is that targets are very likely to reuse `Field`s in V2, whereas in V1 you could only share a field via inheritance.

I'll go with whatever you all prefer, though.